markdown/assignments/sheet05.md

```diff
@@ -1,97 +1,35 @@
-    *   $(w_0,w_1,w_2)^T = (\sqrt{2}, \sqrt{1}, \sqrt{1})^T$
-    *   $ (w_0,w_1,w_2)^T = (-2, -1, -1)^T $
-
-*Thema*: Verständnis Interpretation Perzeptron (Trennebene/Entscheidungsgrenze)
-
-## A6.2: Logische Funktionen als Perzeptron (2P)
-
-*   (1.5P) Das Perzeptron kann zur Ausführung zahlreicher logischer Funktionen verwendet werden. Implementieren Sie die binären Logikfunktionen UND, ODER und KOMPLEMENT und demonstrieren Sie Ihre Implementierung in der Übung/im Praktikum.
-*   (0.5P) Eine grundlegende Einschränkung des Perzeptrons besteht darin, dass es die EXKLUSIV-ODER-Funktion nicht implementieren kann. Erklären Sie den Grund für diese Einschränkung.
-
-*Thema*: Verständnis Perzeptron
-
-
-## A6.3: Perzeptron Lernalgorithmus I (3P)
-
-*   (2P) Implementieren Sie den Perzeptron-Lernalgorithmus. Ihr Programm soll eine Tabelle
-    mit den folgenden Spalten ausgeben, die eine Zeile je Lernschritt enthält:
-    $w_1$, $w_2$, $\theta$, $x_1$, $x_2$, $k$,  $\sum_{i=1}^n w_i x_i - \theta$, vorhergesagte Klasse, Änderung (=,+,-)
-
-    Lernen Sie die boolesche Funktion `OR`. Geben Sie die vollständige Lerntabelle an.
-
-    Die Gewichte $w_1$ und $w_2$ sowie der Schwellenwert $\theta$ sollen jeweils mit dem Wert $0$ initialisiert werden. Die Lernschrittweite $\eta$ sei $0.5$. Nutzen Sie die
-    $\operatorname{sign}$-Funktion als Aktivierungsfunktion.
-
-
-*   (1P) Zeichnen Sie eine geometrische Interpretation des Perzeptrons nach dem Lernen (Trennebene,  Gewichtsvektor, Trainingsbeispiele).
 
-*Thema*: Verständnis Perzeptron und Ablauf Perzeptron-Lernalgorithmus
+## Perzeptron Lernalgorithmus
 
-## A6.4: Lineare Regression und Gradientenabstieg (3P)
+Ziel dieser Aufgabe ist es, mit Hilfe eines Experiments ein Gefühl für die Laufzeit des Perzeptron-Lernalgorithmus zu bekommen und eine Art empirische Approximation zu bestimmen.
 
-Es sind folgende Trainingsdaten gegeben:
+### Datensatz
+*   Konstruieren Sie Ihren eigenen Datensatz $\mathcal{D}$ mit $m=10$ gleichförmig verteilten Zufallspunkten aus dem Bereich $\mathcal{X}=[−1, 1]\times[−1, 1]$.
+*   Wählen Sie auf ähnliche Weise zwei zufällige, gleichmäßig verteilte Punkte aus dem Bereich $[−1, 1]\times[−1, 1]$. Verwenden Sie die Gerade, die durch diese zwei Punkte verläuft, als die Entscheidungsgrenze Ihrer Zielfunktion $f$. Sie können die positiv beschriftete Seite beliebig festlegen.
+*   Werten Sie die Zielfunktion für jeden Datenpunkt $\mathbf{x}^{(j)}$ aus, um die entsprechenden Beschriftungen (Ausgangslabel) $y^{(j)}$ zu erhalten.
 
-$$ ( x^{(1)}, y^{(1)} ) = (1, 1), ( x^{(2)}, y^{(2)} ) = (2, 1), ( x^{(3)}, y^{(3)} ) = (3, 2) $$
+### Training
+Führen Sie nun den Perzeptron Lernalgorithmus $1000$ mal hintereinander aus. Initialisieren Sie jedes mal die Gewichte mit $0$. Wählen Sie in jedem Lernschritt einen Punkt $\mathbf{x}^{(i)}$ *zufällig* aus der Menge der falsch klassifizierten Punkte und aktualisieren Sie die Gewichte entsprechend der folgenden Formel:
```

Comment:
```suggestion
Führen Sie nun den Perzeptron Lernalgorithmus $1000$ mal hintereinander aus. Initialisieren Sie jedesmal die Gewichte mit $0$. Wählen Sie in jedem Lernschritt einen Punkt $\mathbf{x}^{(i)}$ *zufällig* aus der Menge der falsch klassifizierten Punkte und aktualisieren Sie die Gewichte entsprechend der folgenden Formel:
```

Reply:
Habe mal nachgeschaut, anscheinend ist seit 1996 "jedes Mal" die richtige Schreibweise.

Reply:
oh, cool! ich war mir tatsächlich unsicher, ob es groß geschrieben werden sollte wie in "3 Mal"